### ISSUE TITLE
Cadaver Reprocessor not working correctly/ cannot be placed

### ISSUE THREAD

**Issue by SyntheticM4n** [sumghai/Replimat#49]
**Describe the bug:**  

The bug occurs every time I either try to construct the Cadaver Reprocessor or even place it using Debug Mode. It is never actually placed or constructed. If I try to construct it normally it lets me place the template but once the pawns finish building, the template disappears and the actual Cadaver Reprocessor never appears in the world. If I try to place it using debug mode, nothing happens except an error message which I included below.

From my testing this seems to be caused by the recently added 'Holograms' function of Save our Ship 2, which interferes with the compatibility functions of this mod.

**Expected behavior:**  

Placing and using the Cadaver Reprocessor normally.

**Steps to reproduce:**  

Easiest would be:
1. Have Humanoid Alien Races and Save Our Ship 2 active
2. Enable Debug Mode,
3. Select Cadaver Reprocessor,
4. Try and place it anywhere in the game world

**Additional information:**

Root level exception in OnGUI(): System.NullReferenceException: Object reference not set to an instance of an object
  at Replimat.ModCompatibility.AlienCorpseHasOrganicFlesh (Verse.ThingDef def) [0x00028] in <85803c979b184eccbce2ceb84f9af6b4>:0 
  at Replimat.Building_ReplimatCorpseRecycler+<>c.<GetParentStoreSettings>b__21_0 (Verse.ThingDef def) [0x00000] in <85803c979b184eccbce2ceb84f9af6b4>:0 
  at System.Collections.Generic.HashSet`1[T].RemoveWhere (System.Predicate`1[T] match) [0x0003a] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at Replimat.Building_ReplimatCorpseRecycler.GetParentStoreSettings () [0x00018] in <85803c979b184eccbce2ceb84f9af6b4>:0 
  at RimWorld.StorageSettings..ctor (RimWorld.IStoreSettingsParent owner) [0x00010] in <1b9c4c73530448bcbb33283e1d4d7d05>:0 
  at Replimat.Building_ReplimatCorpseRecycler.PostMake () [0x00006] in <85803c979b184eccbce2ceb84f9af6b4>:0 
  at Verse.ThingMaker.MakeThing (Verse.ThingDef def, Verse.ThingDef stuff) [0x000c7] in <1b9c4c73530448bcbb33283e1d4d7d05>:0 
  at RimWorld.Designator_Build.DesignateSingleCell (Verse.IntVec3 c) [0x00081] in <1b9c4c73530448bcbb33283e1d4d7d05>:0 
  at Verse.DesignatorManager.ProcessInputEvents () [0x00058] in <1b9c4c73530448bcbb33283e1d4d7d05>:0 
  at RimWorld.MapInterface.HandleMapClicks () [0x0000f] in <1b9c4c73530448bcbb33283e1d4d7d05>:0 
  at RimWorld.UIRoot_Play.UIRootOnGUI () [0x00093] in <1b9c4c73530448bcbb33283e1d4d7d05>:0 
  at (wrapper dynamic-method) Verse.Root.Verse.Root.OnGUI_Patch1(Verse.Root)
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
(wrapper dynamic-method) Verse.Root:Verse.Root.OnGUI_Patch1 (Verse.Root)

## System and Game Configuration
**Operating System:** Microsoft Windows 10 (64-bit)
**Physical RAM:** 16 GB 
**RimWorld base game version:** 1.3.3076
**RimWorld DLCs loaded:** Royalty and Ideology
**Replimat mod version:** 1.2.7
**Replimat mod downloaded from:** Steam
**Your mod list:** 
HugsLib - 9.0.0
Humanoid Alien Races - 8th August Steam Workshop version
Save Our Ship - 2.5 (7th August Steam Workshop version)
Replimat 1.2.7

## Declaration

I hereby verify that:
- [x] I am running the latest version of the game, Replimat and its dependencies
- [x] I am NOT running any mods with known incompatibilities with Replimat 
- [x] I have narrowed down my modlist to the fewest mods needed to cause this issue
- [x] I have confirmed that the issue does not happen if Replimat is disabled
- [x] I have searched the issue tracker and confirmed no-one else has reported my issue before
- [x] I understand that my issue will be closed and ignored if I do not fill this form properly

**Comment by sumghai** [comment]
Okay, I've reproduced the bug on my end.

I'll have a look at the `AlienCorpseHasOrganicFlesh()` method to see why it's throwing a hissy fit.

**Comment by sumghai** [comment]
@SyntheticM4n - It looks like the latest Steam Workshop version of SOS2 prevents holograms from causing errors with the Cadaver Processor.

However, I'll need to figure out how to blacklist Hologram "corpses" from the Processor's menu, since it doesn't really make sense for holograms to have recyclable corpses.

**Comment by sumghai** [comment]
Should be fixed now.

**Comment by SyntheticM4n** [comment]
Hello. I am not sure if this comment will reach you, but I just wanted to say thank you for taking the time to look into this issue. It is indeed working fine now. I also agree that the holograms should be excluded from the cadaver reporcessor list. Have a good day / night, as I am unsure what time zone you are in.

**Comment by sumghai** [comment]
> Hello. I am not sure if this comment will reach you, but I just wanted to say thank you for taking the time to look into this issue. It is indeed working fine now. I also agree that the holograms should be excluded from the cadaver reporcessor list. Have a good day / night, as I am unsure what time zone you are in. 

No worries, I got your message just before I published the v1.2.8 update, which includes the blacklisting of holograms as discussed.

Enjoy the update!

**Comment by volpol** [comment]
A little update. Setting MAI's flesh to Mechanoid kills its natural healing. I only noticed now because MAI was being healed by the Medpod and currently my colony doesn't have one yet. It might be an overkill anyway as I tried disabling the corresponding patch, thus leaving fleshType as Normal and there were no adverse effects so far. Pawns refuse to haul MAI corpse to the Cadaver Processor so no processing is attempted. Considering how niche this issue seems to be it's fine if you don't invest any more effort into this. Just letting you know.
